### PR TITLE
Add convenience clean/export/install/flash scripts

### PR DESCRIPTION
**Pull request by Brycey92**
### Description

Add convenience clean/export/install/flash scripts

### Test Instructions

Tested all scripts in their respective shells.

### Readiness Checklist
- [ ] I have run `make format` to format the changes
- [x] I have compiled the firmware and the changes have no warnings
- [x] I have compiled the emulator and the changes have no warnings
- [x] I have run `make cppcheck` and checked that `cppcheck_result.txt` has no warnings for the changes
- [x] I have added doxygen comments to any code used by more than one Swadge mode. This includes `/*! \file` comments with Design Philosophy, Usage, and Example sections for new headers.
- [x] I have run `make docs` and checked that `doxy_warnings.txt` has no warnings for the new code

### CODE REVIEW

File: tools/export.ps1 [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+pushd ~/esp/esp-idf
+
+# Install tools
+./install.ps1
+
+# Export paths and variables
+./export.ps1
+
+popd
```
Comment (AEFeinstein) [inline tools/export.ps1:13]: I don't get what this is doing. What's the point of making an `~/esp/` folder and calling a script that may not exist in an empty folder?

File: tools/clean.bat [hunk 2]
```diff
@@ -0,0 +1,5 @@
+make clean
+idf.py clean fullclean
+rmdir /s /q build
+rmdir /s /q spiffs_image
+mkdir spiffs_image
```
Comment (AEFeinstein) [inline tools/clean.bat:5]: This should be a `makefile` target, like `make fullclean`, not a batch file. That way it's cross-platform. This is what you want:

```
fullclean: clean
	idf.py fullclean
	git clean -fX
	-@rm -rf ./docs/html
	-@rm -rf ./spiffs_image/*
	$(MAKE) -C ./tools/sandbox_test clean
	$(MAKE) -C ./tools/hidapi_test clean
	$(MAKE) -C ./tools/bootload_reboot_stub clean
	$(MAKE) -C ./tools/font_maker clean
	$(MAKE) -C ./tools/swadgeterm clean
	$(MAKE) -C ./tools/reboot_into_bootloader clean
	$(MAKE) -C ./tools/spiffs_file_preprocessor clean
```

**Comment by AEFeinstein** [comment]
These scripts may work on your machine, but they're all dependent on an installation in `%UserProfile%\esp\esp-idf\`, which isn't guaranteed.

I'm also not a fan of the one-line scripts (`install.bat`, `export.bat`) that are basically symlinks. Adding `%UserProfile%\esp\esp-idf\` to your path is the cleaner solution. That allows you to call the IDF scripts from anywhere.

`flash.bat` is invalidated by `make usbflash` (try it out!)

File: tools/export.ps1 (hunk 1 above)
Comment (Brycey92) [reply tools/export.ps1:13]: I think I must have stripped down the dev env setup script from last year to make this. Not sure why it does half the stuff it does, so I'll delete it.

**Review by AEFeinstein — APPROVED** [review]

**Pull request merged**